pkg/zkproofs/utils.go

```diff
@@ -0,0 +1,25 @@
+package zkproofs
+
+import (
+	"crypto/rand"
+	"errors"
+
+	"github.com/coinbase/kryptology/pkg/core/curves"
+)
+
+// Generates a non-zero random scalar. The chances of generating a zero scalar are very low.
+func GenerateRandomScalar(curve *curves.Curve) (curves.Scalar, error) {
+	attempts := 0
+	scalar := curve.Scalar.Random(rand.Reader)
```

Comment:
scalar is not updated inside loop

Reply:
alt version
```
// GenerateRandomScalar Generates a non-zero random scalar. The chances of generating a zero scalar are very low.
func GenerateRandomScalar(curve *curves.Curve) (curves.Scalar, error) {
	var scalar curves.Scalar

	for attempts := 0; attempts < 5; attempts++ {
		scalar = curve.Scalar.Random(rand.Reader)
		if !scalar.IsZero() {
			return scalar, nil
		}
	}

	return nil, errors.New("failed to generate a non-zero scalar")
}

Reply:
Thanks for catching this, updated to the clearner version!